src/mprispropertiesadaptor.cpp

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (C) 2015-2023 Jolla Ltd.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#include "mprisplayer_p.h"
+#include "mprispropertiesadaptor_p.h"
+
+using namespace Amber;
+
+/*
+ * Implementation of adaptor class MprisPlayerAdaptor
+ */
+
+MprisPropertiesAdaptor::MprisPropertiesAdaptor(MprisPlayerPrivate *parent)
+    : QDBusAbstractAdaptor(parent)
+    , m_playerPrivate(parent)
+{
+    setAutoRelaySignals(true);
+}
+
+MprisPropertiesAdaptor::~MprisPropertiesAdaptor()
+{
+}
+
+void MprisPropertiesAdaptor::replyPropertyNotFoundError(const QString &interface_name, const QString &property_name)
+{
+    // The property was not found
+    m_playerPrivate->sendErrorReply(QDBusError::UnknownProperty,
+                                    QString::fromLatin1("Property %1%2%3 was not found in object %4")
+                                    .arg(interface_name,
+                                         QString::fromLatin1(interface_name.isEmpty() ? "" : "."),
+                                         property_name, m_playerPrivate->message().path()));
+}
+
+void MprisPropertiesAdaptor::replyPropertyReadOnlyError(const QString &interface_name, const QString &property_name)
+{
+    // The property is read only
+    m_playerPrivate->sendErrorReply(QDBusError::PropertyReadOnly,
+                                    QString::fromLatin1("Property %1.%2 is read-only")
+                                    .arg(interface_name, property_name));
+}
+
+void MprisPropertiesAdaptor::replyInternalError()
+{
+    // The property is read only
+    m_playerPrivate->sendErrorReply(QDBusError::InternalError,
+                                    QString::fromLatin1("Internal error"));
+}
+
+// Format: {DBus property name, MprisPlayer property name or MprisPlayerPrivate getter() method}
+static const QMap<QString, QString> playerGetMap{
```

Comment:
Should this have space before `{`?

Reply:
Honestly I'm not sure... I don't think we have conventions to cover this. I took it to be like an initialiser (see the [cpppreference examples](https://en.cppreference.com/w/cpp/language/list_initialization#Example)), which wouldn't normally attract a space.

To get around the uncertainty, I've added a newline, which I think suits our multiline formatting conventions.